sveltekit-authentication/index.md

```diff
@@ -0,0 +1,449 @@
+
+function getDirectusInstance(fetch,token) {
+
+  const options = fetch ? { globals: { fetch } } : {};
+
+  const directus = createDirectus(PUBLIC_APIURL, options)
+  .with(authentication('cookie', { credentials: 'include' })) // [!code ++]
+  .with(rest());
+
+  if(token) directus.setToken(token); // [!code ++]
+
+  return directus;
+}
+export default getDirectusInstance;
+
+export const constructCookieOpts = (age) => { // [!code ++]
+	return { // [!code ++]
+			'domain': PUBLIC_COOKIE_DOMAIN, // [!code ++]
+			// send cookie for every page // [!code ++]
+			'path': '/', // [!code ++]
+			// server side only cookie so you can't use `document.cookie` // [!code ++]
+			'httpOnly': true, // [!code ++]
+			// only requests from same site can send cookies // [!code ++]
+			// https://developer.mozilla.org/en-US/docs/Glossary/CSRF // [!code ++]
+			'sameSite': "strict", // [!code ++]
+			// only sent over HTTPS in production // [!code ++]
+			'secure': process.env.NODE_ENV === 'production', // [!code ++]
+			// set cookie to expire after a given time // [!code ++]
+			'maxAge': age // [!code ++]
+		} // [!code ++]
+	} // [!code ++]
+```
+
+Additionally we need to define the Domain the Cookie is valid for in the  `.env` file:
+
+```
+PUBLIC_APIURL= "https://directus.example.com"
+PUBLIC_COOKIE_DOMAIN= "example.com" // [!code ++]
+```
+
+## Setup Login/Signup Form
+
+Let's start the user journey from adding a login/signup form:
+
+```svelte [/signin/+page.svelte]
+<script lang="ts">
+	/** @type {import('./$types').PageData} */
+	import { page } from '$app/stores';
+  let email,password;
+	const redirectedFrom = $page.url.searchParams.get('redirectedFrom')
+</script>
+
+<form
+	action="?/login{redirectedFrom?`&redirectedFrom=${redirectedFrom}`:''}"
+	method="POST">
+	<div>
+		<label for="email">Email</label>
+		<input id="email" name="email" type="text" bind:value={email}	
+		/>
+	</div>
+
+	<div>
+		<label for="password">Password</label>
+		<input id="password" name="password" type="password" required bind:value={password}	
+		/>
+	</div>
+
+	<button type="submit">Log in</button>
+	<button formaction="?/register{redirectedFrom?`&redirectedFrom=${redirectedFrom}`:''}">Register</button>
+</form>
+```
+
+The html template is straight forward. We only have a small function to let us redirect the user to a page he was trying to access before needing to login/signup.
+
+The JavaScript form action below will handle the actual request.
+The Directus SDK is not saving HttpOnly Cookies, thus the Directus API is directly accessed from SvelteKit in order to be able to save the tokens in a secure cookie via SvelteKit's cookie handler.
+
+```js [/signin/+page.server.js]
+import { redirect,fail } from '@sveltejs/kit';
+import { PUBLIC_APIURL } from '$env/static/public';
+import { constructCookieOpts } from '$lib/directus';
+
+const REFRESH_TOKEN_EXPIRATION = 30; // in days - adapt this with your Directus Settings
```

Comment:
What does the comment mean?

Reply:
i renamed the variable name and change the comment. it is refering to this setting. https://docs.directus.io/self-hosted/config-options.html#security it should just be in sync with it. for the directus cloud i guess it is 7 days, as stated in the docs. i changed it in the code accordingly to avoid any confusion.